### DISCUSSION TITLE
Suggest to have alility on sqllab to directly consume existing dataset as the data resrouce

### DISCUSSION THREAD

**Discussion by cw1427** [apache/superset#25821] [Ideas]
Background: In many case, for more business scenario, there have more and more complicate sql need to be written, and more tables data need to be union query accross. And we already had designed several business related dataset on superset(a part of sql), and for next coming requirements, we have to organize union query accross multi existing dataset as the raw data source. But the sqllab can only support form facing to exactly datasource but not virtual dataset as the data source.

I know we can manualy copy the existing dataset's specific sql into the same place and doing the re organization, but which would be too long sql, and hard to re-use the existing sql fregment.

Expect: To have a feature to asign some usefuly common dataset(existing with some business scopt purpose sql  that had designed before) to be the superset virtual database table, and to expose them to be accessed directly on sqllab module.

**Comment by rusackas** [comment]
This can become a SIP if you have a technical/design proposal of how to implement this, but for now this sounds like a (potentially plausible) feature request. We'll convert this to a Discussion for now, and hopefully you and/or others can come up with a design/architectural proposal to submit as a SIP for consideration.
